lib/rtp_h264/payloader.ex

```diff
@@ -129,126 +128,94 @@ defmodule Membrane.RTP.H264.Payloader do
+    {[Map.update!(buffer, :payload, &delete_prefix/1)], state}
   end
 
-  defp actions_from_data([payload | rest], acc, state) do
-    {:ok, buffer, state} = buffer_from_payload(payload, state)
-    actions_from_data(rest, [buffer | acc], state)
-  end
+  defp handle_unit_type(:fu_a, buffer, state) do
+    buffers =
+      buffer.payload
+      |> delete_prefix
+      |> FU.fragmentate(state.preferred_size)
+      |> Enum.map(&%Buffer{buffer | payload: &1})
 
-  defp actions_from_data([], acc, state) do
-    {{:ok, [buffer: {:output, Enum.reverse(acc)}]}, state}
+    {buffers, state}
   end
 
-  defp redemand(result) do
-    with {{:ok, actions}, state} <- result |> unify_result do
-      {{:ok, actions ++ [redemand: :output]}, state}
-    end
+  defp handle_unit_type(:stap_a, buffer, state) do
+    state = update_stap_a_properties(delete_prefix(buffer.payload), buffer.metadata, state)
+    {[], state}
   end
 
-  defp buffer_from_payload(payload, state) do
-    state = increment_sequence_number(state)
-    metadata = Map.put(state.metadata, :sequence_number, state.sequence_number)
+  defp delete_prefix(@frame_prefix_longer <> rest), do: rest
+  defp delete_prefix(@frame_prefix_shorter <> rest), do: rest
 
-    buffer = %Buffer{
-      payload: payload,
-      metadata: %{rtp: metadata}
+  defp update_stap_a_properties(
+         <<r::1, nri::2, _type::5, _rest::binary()>> = payload,
+         metadata,
+         %{stap_acc: stap_acc} = state
+       ) do
+    payload = StapA.add_size(payload)
+
+    stap_acc = %{
+      stap_acc
+      | payloads: [payload | stap_acc.payloads],
+        byte_size: stap_acc.byte_size + byte_size(payload),
+        metadata: stap_acc.metadata || metadata,
+        stap_a_reserved: stap_acc.stap_a_reserved * r,
+        stap_a_nri: min(stap_acc.stap_a_nri, nri)
     }
 
-    {:ok, buffer, state}
-  end
-
-  defp increment_sequence_number(state) do
-    %State{state | sequence_number: rem(state.sequence_number + 1, @max_sequence_number + 1)}
+    %{state | stap_acc: stap_acc}
   end
 
-  defp delete_prefix(@frame_prefix_longer <> rest), do: rest
-
-  defp delete_prefix(@frame_prefix_shorter <> rest), do: rest
-
-  defp clear_parser_acc(state) do
-    state
-    |> Map.put(:parser_acc, [])
-    |> Map.put(:acc_byte_size, 0)
-    |> Map.put(:stap_a_reserved, 0)
-    |> Map.put(:stap_a_nri, 0)
-  end
+  defp timestampify(buffers, %{timestamp_resolution: resolution}) do
+    use Ratio
+    alias Membrane.Time
 
-  defp update_stap_a_properties(<<r::1, nri::2, _type::5, _rest::binary()>> = payload, state) do
-    state
-    |> Map.put(:acc_byte_size, state.acc_byte_size + byte_size(payload))
-    |> Map.put(:stap_a_reserved, state.stap_a_reserved && r)
-    |> Map.put(:stap_a_nri, min(state.stap_a_nri, nri))
+    Enum.map(buffers, fn %Buffer{metadata: metadata} = buffer ->
+      rtp_timestamp = Time.to_seconds(resolution * metadata.timestamp)
+      metadata = Map.put(metadata, :rtp, %{timestamp: rtp_timestamp})
```

Comment:
That's not really rtp timestamp as it should start from random value. This is rather offset